crates/lib/Cargo.toml

```diff
@@ -19,6 +19,7 @@ futures = "0.3.24"
 indexmap = "1.9.1"
 lazy_static = "1.4.0"
 log = "0.4.17"
+ndarray = "0.15.6"
```

Comment:
FWIW, I think pulling in `ndarray` is the right choice here, especially if pyQuil already integrates with `numpy`.

The only possible "better" solution I can think of would be to use const generics to return a stack-allocated 2d array. I don't know how well that could be implemented, but it would mean one less dependency for people who don't want to use it.

Reply:
I appreciate that - I went through iterations of rolling our own data structure but nothing felt quite as right as just using `ndarray`. Especially since, as you say, `pyQuil` already integrates with `numpy`. It feels like a bonus to be able to leverage all the linalg capabilities of the crate as well.